opm/input/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -68,17 +68,28 @@
 #include <fmt/format.h>
 
 namespace {
-    Opm::Box makeGlobalGridBox(const Opm::EclipseGrid* gridPtr)
+    Opm::Box makeGlobalGridBox(const Opm::EclipseGrid* gridPtr,
+                               const std::vector<int>* actnum = nullptr)
     {
         return Opm::Box {
             *gridPtr,
-            [gridPtr](const std::size_t global_index)
+            [gridPtr, actnum](const std::size_t global_index)
             {
-                return gridPtr->cellActive(global_index);
+                if (!actnum) {
+                    return gridPtr->cellActive(global_index);
+                }
+                if (actnum->empty()) {
+                    return true;
+                } else if (global_index >= actnum->size()) {
+                    return false;
+                } else {
+                    return (*actnum)[global_index] > 0;
+                }
             },
-            [gridPtr](const std::size_t global_index)
+            [gridPtr, actnum](const std::size_t global_index) -> std::size_t
             {
-                return gridPtr->activeIndex(global_index);
+                return actnum ? (*actnum)[global_index]
+                              : gridPtr->activeIndex(global_index);
             }
```

Comment:
This function is supposed to return the index&ndash;in the range `0..#active-1`&ndash;of an active `global_index` cell.  As written, this will only ever return `1` if `actnum != nullptr`.

Reply:
doh. yeah, a bit too fast here. obviously i need a translation to indices which will be costly to do here.

Reply:
i've added a mapping now but i doubt this is a good approach. i can't find an existing mapping from cart to compressed, only compressed to cart so i have to invert. i've prioritized memory usage over performance, ie, it uses an unordered_map not a flat vector. right now it's kinda stupid though as forms the flat vector to make the map, i can fix that if this approach is the way to go..